app/controllers/api/v1/cookie_and_jwt_api/protocol_subscriptions_controller.rb

```diff
@@ -25,6 +39,29 @@ def set_protocol_subscription
 
           render(status: :not_found, json: 'Protocol subscription met dat ID niet gevonden')
         end
+
+        def start_date
+          return start_time if protocol_subscription_create_params[:start_date].blank?
+
+          Time.zone.parse(protocol_subscription_create_params[:start_date])
+        end
+
+        def start_time
+          return Time.zone.now if no_start_time_given
+
+          next_midnight = TimeTools.increase_by_duration(Time.zone.now.beginning_of_day, 1.day)
+          TimeTools.increase_by_duration(next_midnight, protocol_subscription_create_params[:start_time].to_i)
+        end
+
+        def no_start_time_given
+          protocol_subscription_create_params[:start_time].blank? ||
+            (protocol_subscription_create_params[:start_time].to_i.zero? &&
```

Comment:
why is this? This seems a bit weird, you do it .to_i and check if it is 0, but it cannot be '0'? Perhaps add a comment why you do it like this.

Reply:
It's because .to_i on a random string (or empty string) simply returns 0 instead of raising an error. I'll add a comment.

![ConEmu64_2020-03-30_18-36-15](https://user-images.githubusercontent.com/607965/77937961-580b3d00-72b5-11ea-8978-f372660a4c05.png)

Reply:
Fixed.